### ISSUE TITLE
Another Style/FormatStringToken false positive

### ISSUE THREAD

**Issue by mikegee** [bbatsov/rubocop#5630]
Github didn't let @lionel218 open this issue for some reason. See the original report at https://github.com/bbatsov/rubocop/issues/5245#issuecomment-370500831

--------

## Expected behavior

no offense from `Style/FormatStringToken`

## Actual behavior

```
% bundle exec bin/rubocop test/test.rb --only Style/FormatStringToken
Inspecting 1 file
C

Offenses:

test/test.rb:2:75: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
  format('%<day>s %<start>s-%<end>s', day: open_house.starts_at.strftime('%a'),
                                                                          ^^
test/test.rb:4:76: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
                                      end: open_house.ends_at.strftime('%l %p').strip)
                                                                           ^^

1 file inspected, 2 offenses detected
```

## Steps to reproduce the problem

Use this file:

```ruby
def event_date(open_house)
  format('%<day>s %<start>s-%<end>s', day: open_house.starts_at.strftime('%a'),
                                      start: open_house.starts_at.strftime('%l'),
                                      end: open_house.ends_at.strftime('%l %p').strip)
end
```

## RuboCop version

```
$ rubocop -V
0.53.0 (using Parser 2.5.0.2, running on ruby 2.5.0 x86_64-darwin16)
```

**Comment by lionel218** [comment]
Did you solve this issue?

**Comment by katherine-boost** [comment]
Just upgraded to `0.75` and I'm seeing this error again. Anyone else?

**Comment by buehmann** [comment]
@katherine-boost I was working on this cop for the 0.75 release. What error exactly are you seeing?

**Comment by katherine-boost** [comment]
The error:
```
Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
```

The line of code provoking the error:
```ruby
formatted_cents = format('%02d', cents % 100)
```

**Comment by buehmann** [comment]
@katherine-boost That is not an error. The cop behaves as designed. With style `annotated`, which is the default, it suggests to use named format sequences and not positional ones. In your example:
```ruby
formatted_cents = format('%02<remainder>d', remainder: cents % 100)
```
Also see https://docs.rubocop.org/en/stable/cops_style/#styleformatstringtoken

**Comment by ndbroadbent** [comment]
Hello, just wanted to post another issue in here. When I set this in my `rubocop.yml` to enforce the template style:

```
Style/FormatStringToken:
  EnforcedStyle: template
```

I'm getting this error:

```
C: Style/FormatStringToken: Prefer template tokens (like %{foo}) over annotated tokens (like %<foo>s).
  format('%<red>02x%<green>02x%<blue>02x', color_values)
```

I'm not sure if I am able to switch these to the template style, because I believe that is only for strings, while I need to use the `%02x` to get hex values.

**Comment by buehmann** [comment]
@ndbroadbent Good point. Would you mind creating a new issue for this? Template style has indeed no way I know to influence the type conversion. By the way, it seems to accept non-strings as an argument, but does not simply do `to_s` (e.g. for arrays).
